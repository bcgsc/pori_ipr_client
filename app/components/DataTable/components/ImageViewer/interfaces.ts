interface SelectedRow {
  image: ImageData,
}

interface ImageData {
  caption?: string | null,
  createdAt?: string,
  data: string,
  filename?: string,
  format: string,
  ident?: string,
  key: string,
  title?: string | null,
  updatedAt?: string | null,
}

export {
  ImageData,
  SelectedRow,
};
