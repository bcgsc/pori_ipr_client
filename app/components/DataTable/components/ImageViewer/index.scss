.dialog {
  &__image {
    margin: 0 auto;
  }
}