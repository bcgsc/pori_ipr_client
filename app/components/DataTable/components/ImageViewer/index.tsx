import React from 'react';
import {
  Dialog,
  DialogContent,
  DialogActions,
  Button,
} from '@mui/material';
import Image from '@/components/Image';
import { SelectedRow } from './interfaces';

import './index.scss';

interface Props {
  /** Handles dialog close */
  onClose: () => void;
  /** Row object selected from table */
  selectedRow: SelectedRow,
  /** Dialog open state */
  isOpen: boolean;
}

const ImageViewer = ({ onClose, selectedRow, isOpen }: Props): JSX.Element => {
  const handleClose = (): void => {
    onClose();
  };

  return (
    <Dialog
      onClose={handleClose}
      open={isOpen}
      maxWidth="xl"
    >
      <DialogContent>
        {selectedRow.image && (
          <Image
            image={selectedRow.image}
            showTitle
            showCaption
            isZoomable={false}
          />
        )}
      </DialogContent>
      <DialogActions>
        <Button color="primary" onClick={handleClose}>
          Close
        </Button>
      </DialogActions>
    </Dialog>
  );
};

export default ImageViewer;
