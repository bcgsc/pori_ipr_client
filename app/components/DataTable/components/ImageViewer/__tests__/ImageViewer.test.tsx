import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import ImageViewer from '..';

const mockSelectedRow = {
  image: {
    caption: 'here beith thy image',
    createdAt: '20211-20-01T12:20:10.485Z',
    data: 'dbf2003660cb',
    filename: 'testingimage.png',
    format: 'PNG',
    ident: 'ab502bf-077fbba-0019-9ade',
    key: 'testing.image',
    title: 'Testing Image',
    updatedAt: '20211-20-01T12:20:10.485Z',
  },
};

describe('ImageViewer', () => {
  test('It matches the snapshot', () => {
    const { asFragment } = render(
      <ImageViewer
        onClose={() => {}}
        selectedRow={mockSelectedRow}
        isOpen
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('Clicking the close button calls onClose', async () => {
    const mockOnClose = jest.fn();
    render(
      <ImageViewer
        onClose={mockOnClose}
        selectedRow={mockSelectedRow}
        isOpen
      />,
    );
    fireEvent.click(await screen.findByRole('button', { name: 'Close' }));
    expect(mockOnClose).toHaveBeenCalledTimes(1);
    expect(mockOnClose).toHaveBeenCalledWith();
  });
});
