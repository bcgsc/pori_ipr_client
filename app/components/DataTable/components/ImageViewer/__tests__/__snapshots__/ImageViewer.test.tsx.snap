// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageViewer It matches the snapshot 1`] = `<DocumentFragment />`;
